editor/shader_create_dialog.cpp

```diff
@@ -390,16 +388,25 @@ String ShaderCreateDialog::_validate_path(const String &p_path, bool p_file_must
 
 	ShaderTypeData data = language_data[language_menu->get_selected()];
 
+	bool found = false;
 	bool match = false;
-	int index = 0;
+
 	for (Set<String>::Element *E = extensions.front(); E; E = E->next()) {
 		if (E->get().nocasecmp_to(extension) == 0) {
-			match = true;
+			found = true;
+			for (List<String>::Element *E2 = language_data[current_language].extensions.front(); E2; E2 = E2->next()) {
```

Comment:
This can use a range iterator now:
```cpp
for (const String &lang_ext : language_data[current_language]) {
	if (lang_ext.nocasecmp_to(extension == 0) { ... }
```

Same with the previous `for (Set<String>::...)`.

Reply:
There's a handful of other `for` loops with `List` or `Set` in this file which can be ported to range iterators (see #50284).

Reply:
I think I'm done with it